torchaudio/prototype/functional/functional.py

```diff
@@ -134,7 +134,9 @@ def convolve(x: torch.Tensor, y: torch.Tensor, mode: str = "full") -> torch.Tens
     return _apply_convolve_mode(result, x_size, y_size, mode)
 
 
-def add_noise(waveform: torch.Tensor, noise: torch.Tensor, lengths: torch.Tensor, snr: torch.Tensor) -> torch.Tensor:
+def add_noise(
+    waveform: torch.Tensor, noise: torch.Tensor, snr: torch.Tensor, lengths: Optional[torch.Tensor] = None
```

Comment:
shoud it be `length` as `waveform` and `noise` are both singular?

Reply:
i'd stick with `lengths` because, whereas a waveform and noise can comprise multiple components, "length" represents only one quantity

this is also consistent with other library components, e.g. rnn-t loss, conformer, emformer

Reply:
makes sense. let's also use `lengths` in the future functions and modules then.